### PR TITLE
Replace starboard/common/log with base/logging in the common source files

### PR DESCRIPTION
**Pull request by iuriionishchenko**
Replace starboard/common/log with base/logging in the common source files

b/322248899

Change-Id: Idf38e7e48f7583778b1a06e069926b6427272af2

### CODE REVIEW

File: v8/src/base/platform/platform-starboard.cc [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include <stdio.h>
 #include <unistd.h>
 
+#include "base/logging.h"
```
Comment (kaidokert) [inline v8/src/base/platform/platform-starboard.cc:12]: v8 is actually an exception. V8 does not directly depend on Chromium base// ( v8 can be built completely standalone ). 

So this change should be undone.

File: third_party/skia/src/core/SkMatrix.cpp [hunk 2]
```diff
@@ -1533,7 +1534,7 @@ template <MinMaxOrBoth MIN_MAX_OR_BOTH> bool get_scale_factor(SkMatrix::TypeMask
           if (kBoth_MinMaxOrBoth == MIN_MAX_OR_BOTH)  {
             results[1] = 0;
           }
-          SB_DLOG(WARNING) << "Unable to calculate scale factor of a matrix.";
+          DLOG(WARNING) << "Unable to calculate scale factor of a matrix.";
```
Comment (kaidokert) [inline third_party/skia/src/core/SkMatrix.cpp:1537]: Skia code should use `SkDebugf`

File: glimp/egl/display.cc [hunk 3]
```diff
@@ -20,6 +20,7 @@
 #include <utility>
 #include <vector>
 
+#include "base/logging.h"
```
Comment (kaidokert) [inline glimp/egl/display.cc:23]: Glimp does not depend on Chrome //base, so changes in //glimp should be undone. It's a lower layer standalone library - it's okay here to use Starboard directly.

File: third_party/zlib/google/compression_utils_unittest.cc [hunk 4]
```diff
@@ -107,7 +107,7 @@ TEST(CompressionUtilsTest, OutputCompressionAndDecompressionDuration) {
   std::string compressed_data;
```
Comment (kaidokert) [inline third_party/zlib/google/compression_utils_unittest.cc:107]: Please remove `#include "starboard/common/log.h"` at the top of the file as well, if possible.

File: v8/src/base/platform/platform-starboard.cc (hunk 1 above)
Comment (iuriionishchenko) [reply v8/src/base/platform/platform-starboard.cc]: Done

File: glimp/egl/display.cc (hunk 3 above)
Comment (iuriionishchenko) [reply glimp/egl/display.cc]: Done

File: third_party/skia/src/core/SkMatrix.cpp (hunk 2 above)
Comment (iuriionishchenko) [reply third_party/skia/src/core/SkMatrix.cpp]: Done

File: third_party/zlib/google/compression_utils_unittest.cc (hunk 4 above)
Comment (iuriionishchenko) [reply third_party/zlib/google/compression_utils_unittest.cc:106]: done

File: v8/src/base/platform/platform-starboard.cc [hunk 5]
```diff
@@ -17,8 +17,8 @@
 #include "src/base/timezone-cache.h"
 #include "src/base/utils/random-number-generator.h"
 #include "starboard/client_porting/eztime/eztime.h"
-#include "starboard/common/condition_variable.h"
 #include "starboard/common/log.h"
+#include "starboard/common/condition_variable.h"
```
Comment (kaidokert) [inline v8/src/base/platform/platform-starboard.cc:21]: This change seems not needed here now.

File: third_party/skia/src/core/SkMatrix.cpp [hunk 6]
```diff
@@ -5,6 +5,7 @@
  * found in the LICENSE file.
  */
 
+#include "base/logging.h"
```
Comment (kaidokert) [inline third_party/skia/src/core/SkMatrix.cpp:8]: This include should not be needed now.

File: glimp/egl/surface.cc [hunk 7]
```diff
@@ -19,6 +19,7 @@
 
 #include <utility>
 
+#include "base/logging.h"
```
Comment (kaidokert) [inline glimp/egl/surface.cc:22]: All of //glimp does not (and should not ) depend on //base, so please undo all changes in this directory.

Glimp component is expected to depend on Starboard itself only. We are actually planning to move the entire component under //starboard/shared/glimp in future.

**Review by kaidokert — CHANGES_REQUESTED** [review]
Changes under //base and //cobalt look okay to me.

Please back out //glimp changes, and follow up on the rest of the comments in //third_party/ and //v8 too, and after that this should be good to go in.

File: v8/src/base/platform/platform-starboard.cc (hunk 5 above)
Comment (iuriionishchenko) [reply v8/src/base/platform/platform-starboard.cc]: All changes in the //v8 were reverted

File: glimp/egl/surface.cc (hunk 7 above)
Comment (iuriionishchenko) [reply glimp/egl/surface.cc]: All changes in the //glimp were reverted

File: third_party/skia/src/core/SkMatrix.cpp (hunk 6 above)
Comment (iuriionishchenko) [reply third_party/skia/src/core/SkMatrix.cpp]: All changes in the //third_party were reverted

**Review by kaidokert — APPROVED** [review]

File: cobalt/browser/memory_settings/table_printer_test.cc [hunk 8]
```diff
@@ -16,6 +16,7 @@
 
 #include "cobalt/browser/memory_settings/table_printer.h"
 
+#include "base/logging.h"
 #include "starboard/common/log.h"
```
Comment (kaidokert) [inline cobalt/browser/memory_settings/table_printer_test.cc:20]: Nit: `#include starboard/common/log.h` can be removed here too now, i think.

File: cobalt/watchdog/watchdog.cc [hunk 9]
```diff
@@ -24,6 +24,7 @@
 #include "base/command_line.h"
 #include "base/json/json_reader.h"
 #include "base/json/json_writer.h"
+#include "base/logging.h"
 #include "starboard/common/file.h"
 #include "starboard/common/log.h"
```
Comment (kaidokert) [inline cobalt/watchdog/watchdog.cc:29]: Nit: `#include starboard/common/log.h` can be removed too.

File: cobalt/browser/memory_settings/table_printer_test.cc (hunk 8 above)
Comment (iuriionishchenko) [reply cobalt/browser/memory_settings/table_printer_test.cc]: Removed

File: cobalt/watchdog/watchdog.cc (hunk 9 above)
Comment (iuriionishchenko) [reply cobalt/watchdog/watchdog.cc]: Removed

**Review by amurovanyi — APPROVED** [review]

**Pull request merged**